packages/gitlab/src/api/GitlabCIClient.ts

```diff
@@ -146,21 +149,54 @@ export class GitlabCIClient implements GitlabCIApi {
     }
 
     async getPipelineSummary(
-        projectID?: string | number
+        projectID?: string | number,
+        refList?: string[]
     ): Promise<PipelineSchema[] | undefined> {
-        const [pipelineObjects, projectObj] = await Promise.all([
-            this.callApi<PipelineSchema[]>(
+        if (!refList || refList.length === 0) {
+            return this.callApi<PipelineSchema[]>(
                 'projects/' + projectID + '/pipelines',
                 {}
-            ),
-            this.callApi<Record<string, string>>('projects/' + projectID, {}),
-        ]);
+            );
+        }
+
+        const projectObj = await this.callApi<Record<string, string>>(
+            'projects/' + projectID,
+            {}
+        );
+
+        const pipelineObjects = [];
+        let page = 1;
+        let response;
+        do {
+            response = await this.callApi<PipelineSchema[]>(
```

Comment:
You cannot crawl all pipelines because it can be very inefficient. I suggest using [branches](https://docs.gitlab.com/ee/api/branches.html) API, to get all eligible branches and then query the pipelines API with the right branches.

Reply:
'Branches' API has the qs parameter that is very useful to implement the wildcard.

Reply:
Can do. How do you recommend configuring the plugin so that you can test the backend against a real GitLab instance locally? I've been running `yarn start` which only tests the repo against the mock response with mock queries.

Reply:
Yes, it is hard to test with a real GitLab instance; I have to work on it to increase DevEx. I usually test it by integrating it into a pre-configured backstage environment configured with `gitlab.com` and I usually link the library using yarn.